### DISCUSSION TITLE
code-block-label or code-block-id

### DISCUSSION THREAD

**Discussion by camilogarciabotero** [quarto-dev/quarto-cli#623] [Feature Requests]
Hi all,

Thank you for working so hard on the awesome Quarto. I was playing around with it and wondered if there was a feature to label or mark the code blocks after the language it displays. Here is an example of an `R` code block. 

![image](https://user-images.githubusercontent.com/25586362/163067731-866a13ea-cfeb-45c9-a714-f7f21186a73d.png)

Cheers.

**Comment by jjallaire** [comment]
No we don't currently have this option but will definitely consider adding it!

**Comment by jamespeapen** [comment]
I don't have much JS experience so this is hacky, but I've got this script embedded in my quarto documents. It finds the code block language string and adds a `<p>` containing that language label. It sets the label color to the color of the current code highlighting theme's comment color.

`.qmd` file

```
text: <script type="text/javascript" src="lang-tag.js"></script>
```

`lang-tag.js`

```js
document.addEventListener("DOMContentLoaded", function() {
    var codeBlocks = document.querySelectorAll("pre.sourceCode");
    codeBlocks.forEach(function(codeBlock) {
    var langTag = document.createElement('p');
    langTag.className = "lang-tag";
    langTag.textContent = codeBlock.className.split(' ')[1];
    codeBlock.parentNode.insertBefore(langTag, codeBlock);
    });
})
```

`styles.scss`

```css
div p.lang-tag {
    margin: 0.2rem;
    text-align: left;
    font-size: 0.8rem;
    font-weight: bolder;
    color: var(--quarto-hl-co-color);
}
```


![image](https://github.com/quarto-dev/quarto-cli/assets/39574560/1d862977-22a2-4795-b3a6-8d11479b5a95)

Happy to work on a PR for this, but would need some pointers on integrating this into quarto.
